### DISCUSSION TITLE
Update remote entity to show available activities

### DISCUSSION THREAD

**Discussion by bdraco** [home-assistant/architecture#526] [Entity Models] [answered]
The remote entity model already supports selecting an activity for `async_turn_off` `async_turn_on`, and `async_toggle`, however there is no standard way to show the available activities or the current activity.

This proposal is to add two new properties

`current_activity` - The currently active activity
`activity_list` - The list of available activities

While Logitech currently dominates the connected remote market, this approach should work for new players that enter the market.

We should be able to offer a better experience in HomeKit and other services with this change.

Core PR https://github.com/home-assistant/core/pull/47237
Frontend PR https://github.com/home-assistant/frontend/pull/8506
Dev Docs: https://github.com/home-assistant/developers.home-assistant/pull/827
HomeKit Update PoC: https://github.com/home-assistant/core/pull/47261

**Comment by elupus** [comment]
How is an activity related to commands? The broadlink remote control supports learned IR codes and commands. Are we expecting all learned commands to show up in the activity list?

**Reply by bdraco** [reply]
Activities are a predefined set of commands to run to achieve a state.  The activity is defined on the remote itself and not by Home Assistant.  Learning and deleting commands wouldn’t be inside the scope of this

**Reply by elupus** [reply]
But is a keypress an activity?

**Reply by bdraco** [reply]
> But is a keypress an activity?

An activity looks something like:

Turn on TV
Turn on Cable Box
Turn on Amp
Set TV to HDMI1
Set Amp to Cable Box
Set Cable Box to Cable

While they might simulate a key press to accomplish this, some of them could be network actions, ir, rf, etc.

**Reply by elupus** [reply]
Right, but what I'm asking from an architectural perspective is, should for example the broadlink or my new philips tv integration expose it's keypresses in this new list?

In other words. Are we equating this list with a generic command.

I absolutely see a benefit to doing that. It would make the keypresses a bit easier to use from a standard GUI. But if they are conceptually different from activities we probably should not.

Ps. I ignore the concept of learning keys/activities here.

**Reply by bdraco** [reply]
This would only be for devices that supported predefined activities or macros.

In the future, I think it would be great to provide a standardized key mapping for learning play/pause/etc so we can map these to other integrations, but this is outside the scope of this change

**Reply by elupus** [reply]
Ok that is perfectly fine. We should make sure this is defined in the dev docs i think.

**Reply by bdraco** [reply]
I updated the developer docs to document the existing `activity` param for `turn_on`, `turn_off`, and `toggle` in https://github.com/home-assistant/developers.home-assistant/pull/827

**Comment by balloob** [comment, marked as answer]
This makes sense.

**Comment by frenck** [comment]
I like the suggestion, I do wonder if we should stick more towards the Home Assistant naming styles. In most places, we call this presets?

I'm not sure if we should introduce a different naming here, as, at this point, that is highly specific to Logitech.

**Reply by bdraco** [reply]
I went with `activity` since it was already present for the `async_turn_off` `async_turn_on`, and `async_toggle` services since 2016 and I wanted to avoid a breaking change.

**Reply by frenck** [reply]
Alright!
